### PR TITLE
Make container always restart if it stops

### PR DESCRIPTION
**Pull request by longkb**
Signed-off-by: Kim Bao Long <longkb@vn.fujitsu.com>

### CODE REVIEW

File: kinder/pkg/build/alter/alter.go [hunk 1]
```diff
@@ -327,6 +327,7 @@ func (c *Context) createAlterContainer(bc *bits.BuildContext) (id string, err er
 		"-v", fmt.Sprintf("%s:%s", bc.HostBasePath(), bc.ContainerBasePath()),
 		// the container should hang forever so we can exec in it
 		"--entrypoint=sleep",
+		"--restart=always", //Always restart the container if it stops
```
Comment (neolit123) [inline kinder/pkg/build/alter/alter.go]: thank you for the PR.

there should be a space after `//` and before `Always`.

but to ask the more important questions, can you explain why this is needed?
are you seeing cases where we need to restart the alter container?

Comment (longkb) [reply kinder/pkg/build/alter/alter.go]: No, I didn't see any errors.  My PR aims to ensure redundancy for container in these case of it is stopped or got an errors.

**Comment by neolit123** [comment]
@longkb 

the CI seems to have failed on this change:
> time="09:15:18" level=debug msg="Running: /usr/bin/docker [docker run -d -v /tmp/kinder-alter-image907514342:/alter --entrypoint=sleep --restart always --name=kind-build-2b112d59-f774-4d9d-aa46-23e487897ddb kindest/base:v20190403-1ebf15f infinity]"
time="09:15:18" level=error msg="unknown flag: --restart always"
time="09:15:18" level=error msg="See 'docker run --help'."
time="09:15:18" level=error msg="Image alter Failed! failed to create alter container: exit status 125"
time="09:15:18" level=debug msg="Running: [docker rm -f -v kind-build-2b112d59-f774-4d9d-aa46-23e487897ddb]"
Error: error altering node image: failed to create alter container: exit status 125
 exit status 1

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubeadm/2043/pull-kubeadm-kinder-upgrade-master/1233318839840673792/build-log.txt

have you tested it locally?

**Comment by longkb** [comment]
> 
> 
> @longkb
> 
> the CI seems to have failed on this change:
> 
> > time="09:15:18" level=debug msg="Running: /usr/bin/docker [docker run -d -v /tmp/kinder-alter-image907514342:/alter --entrypoint=sleep --restart always --name=kind-build-2b112d59-f774-4d9d-aa46-23e487897ddb kindest/base:v20190403-1ebf15f infinity]"
> > time="09:15:18" level=error msg="unknown flag: --restart always"
> > time="09:15:18" level=error msg="See 'docker run --help'."
> > time="09:15:18" level=error msg="Image alter Failed! failed to create alter container: exit status 125"
> > time="09:15:18" level=debug msg="Running: [docker rm -f -v kind-build-2b112d59-f774-4d9d-aa46-23e487897ddb]"
> > Error: error altering node image: failed to create alter container: exit status 125
> > exit status 1
> 
> https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubeadm/2043/pull-kubeadm-kinder-upgrade-master/1233318839840673792/build-log.txt
> 
> have you tested it locally?

Hmm, I have tried in my local, it still runs. hmmm

**Comment by longkb** [comment]
@neolit123 : This is trivial fix. If it cannot pass the CI, let's close this PR :)

**Pull request closed without merging**